server/controllers/user.js

```diff
@@ -10,40 +9,40 @@ dotenv.config();
 const app = express();
 app.use(express.json);
 
-export const usersSignUp = (req, res) => {
-  bcrypt.hash(req.body.password, 10, async (err, hash) => {
-    class User {
-      constructor() {
-        this.firstName = req.body.firstName,
-        this.lastName = req.body.lastName,
-        this.email = req.body.email,
-        this.password = hash,
-        this.address = req.body.address,
-        this.bio = req.body.bio,
-        this.occupation = req.body.occupation,
-        this.expertise = req.body.expertise;
-      }
+export const usersSignUp = async (req, res) => {
+  const hashedPassword = await bcrypt.hash(req.body.password, 10);
+  class User {
+    constructor() {
+      this.firstName = req.body.firstName,
+      this.lastName = req.body.lastName,
+      this.email = req.body.email,
+      this.password = hashedPassword,
+      this.address = req.body.address,
+      this.bio = req.body.bio,
+      this.occupation = req.body.occupation,
+      this.expertise = req.body.expertise;
     }
-    const newUser = new User();
+  }
 
+  const newUser = new User();
 
-    const insert = 'INSERT INTO users(firstName, lastName, email, password, address, bio, occupation, expertise) VALUES($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *';
-    const { rows } = await pool.query(insert,
-      [newUser.firstName, newUser.lastName, newUser.email, newUser.password, newUser.address, newUser.bio, newUser.occupation, newUser.expertise]);
-    const token = encrypter(rows[0].email, rows[0].userid);
 
-    return res.status(201).json({
-      status: 201,
-      message: 'User created successfully',
-      data: {
-        token,
-      },
-    });
+  const insert = 'INSERT INTO users(firstName, lastName, email, password, address, bio, occupation, expertise) VALUES($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *';
+  const { rows } = await pool.query(insert,
+    [newUser.firstName, newUser.lastName, newUser.email, newUser.password, newUser.address, newUser.bio, newUser.occupation, newUser.expertise]);
+  const token = encrypter(rows[0].email, rows[0].userid);
+
+  return res.status(201).json({
+    status: 201,
+    message: 'User created successfully',
+    data: {
+      token,
+    },
   });
 };
-export const usersSignIn = (req, res) => {
-  const user = users.find((usr) => usr.email === req.body.email);
-  const token = encrypter(user.email);
+export const usersSignIn = async (req, res) => {
+  const { rows } = await pool.query(`SELECT userid, email FROM users WHERE email='${req.body.email}';`);
```

Comment:
@gadishimwe , all your SQL queries should be in a helper file.. you should not write SQL queries in the controller. please sync with your teammates who have implemented this.. and remember. do not over collaborate